pkg/adapter/adapter.go

```diff
@@ -152,56 +156,38 @@ func (a *Adapter) PollForMessages(stopCh <-chan struct{}) error {
 		go a.processMessages(wg, workerQueue)
 	}
 
-	retryChan := make(chan bool)
-	defer a.rmqHelper.CleanupRabbitMQ(a.connection, logger)
 	for {
-		a.connection, a.channel, err = a.rmqHelper.SetupRabbitMQ(rabbit.VHostHandler(a.config.RabbitURL, a.config.Vhost), rabbit.ChannelQoS, logger)
-		if err != nil {
-			logger.Errorf("error creating RabbitMQ connections: %s, waiting for a retry", err)
-		} else {
-			queue, err = a.channel.QueueInspect(a.config.QueueName)
-			if err != nil {
-				logger.Error(err.Error())
-				continue
+		if a.rmqHelper.GetChannel() != nil {
+			queue, err = a.rmqHelper.GetChannel().QueueInspect(a.config.QueueName)
+			if err == nil {
+				msgs, err = a.ConsumeMessages(&queue, logger)
+				if err == nil {
+					for {
+						select {
+						case <-stopCh:
+							close(workerQueue)
+							wg.Wait()
+							logger.Info("Shutting down...")
+							return nil
+						case msg, ok := <-msgs:
+							if !ok {
+								close(workerQueue)
+								wg.Wait()
+								return amqp.ErrClosed
+							}
+							workerQueue <- msg
+						}
+					}
+				}
 			}
-			msgs, _ = a.ConsumeMessages(&queue, logger)
-			go PollCycle(retryChan, stopCh, workerQueue, wg, msgs, a.rmqHelper, logger)
-		}
-		if retry := a.rmqHelper.WaitForRetrySignal(); !retry {
-			logger.Warn("stopped listenning for RabbitMQ resources retries")
-			return nil
+		} else {
+			err = amqp.ErrClosed
 		}
 		if err != nil {
-			retryChan <- true
-		}
-		logger.Warn("recreating RabbitMQ resources")
-	}
-}
-
-func PollCycle(
-	retryChan chan bool,
-	stopCh <-chan struct{},
-	workerQueue chan amqp.Delivery,
-	wg *sync.WaitGroup,
-	msgs <-chan amqp.Delivery,
-	rmqHelper rabbit.RabbitMQHelperInterface,
-	logger *zap.SugaredLogger) {
-	for {
-		select {
-		case <-stopCh:
-			close(workerQueue)
-			wg.Wait()
-			logger.Info("Shutting down...")
-			rmqHelper.SignalRetry(false)
-			return
-		case msg, ok := <-msgs:
-			if !ok {
-				return
-			}
-			workerQueue <- msg
-		case <-retryChan:
-			return
+			logger.Error(err)
+			logger.Info("Recreating RabbitMQ Connection and Channel")
 		}
+		time.Sleep(time.Second)
```

Comment:
just a nit: for readability, this sleep can be inside of the `if err != nil` block. There isn't a non-error case where we'd be sleeping. If it reaches this point, it's because an error happened.